Dockerfile

```diff
@@ -1,29 +1,30 @@
-FROM golang:1.13 as secrets-provider-builder
+FROM golang:1.13 as secrets-provider-builder-base
 MAINTAINER CyberArk Software Ltd.
 
 ENV GOOS=linux \
     GOARCH=amd64 \
     CGO_ENABLED=0
 
-COPY . /opt/secrets-provider-for-k8s
+RUN apt-get update
+
+RUN go get -u github.com/jstemmer/go-junit-report && \
+    go get github.com/smartystreets/goconvey
 
 WORKDIR /opt/secrets-provider-for-k8s
 
 EXPOSE 8080
 
-RUN apt-get update && apt-get install -y jq
+COPY go.mod .
+COPY go.sum .
 
-RUN go get -u github.com/jstemmer/go-junit-report && \
-    go get github.com/smartystreets/goconvey
-
-RUN go build -a -installsuffix cgo -o secrets-provider ./cmd/secrets-provider
+RUN go mod download
 
 # =================== BUSYBOX LAYER ===================
 # this layer is used to get binaries into the main container
 FROM busybox
 
-# =================== MAIN CONTAINER ===================
-FROM alpine:3.11 as secrets-provider
+# =================== BASE MAIN CONTAINER ===================
```

Comment:
can you please add a comment on why we need a base main container and base builder?